python/cog/server/runner.py

```diff
@@ -206,13 +211,13 @@ def _upload_files(self, output: Any) -> Any:
             return output
 
         try:
-            # FIXME: clean up output files
+            # TODO: clean up output files
             return self._file_uploader(output)
         except Exception as error:
-            self.failed(error)
-            # re-raise and crash the container
-            # FIXME: ensure director gracefully handles this container crashing
-            raise error
+            # If something goes wrong uploading a file, it's irrecoverable.
+            # The re-raised exception will be caught and cause the prediction
+            # to be failed, with a useful error message.
+            raise FileUploadError("Got error trying to upload output files") from error
 
 
 def predict(
```

Comment:
I think my preference here might be to wrap the existing code up as `_predict` and then have `predict` be an error handling wrapper around `_predict` -- that might make it easier to verify that all errors are caught and handled appropriately, rather than indenting the entire function body.

Reply:
Ah yeah, that's much nicer, thanks